operator/api/v1alpha1/keptnappversion_conversion.go

```diff
@@ -0,0 +1,227 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"github.com/keptn/lifecycle-toolkit/operator/api/v1alpha1/common"
+	"github.com/keptn/lifecycle-toolkit/operator/api/v1alpha2"
+	v1alpha2common "github.com/keptn/lifecycle-toolkit/operator/api/v1alpha2/common"
+	"go.opentelemetry.io/otel/propagation"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts the src v1alpha1.KeptnAppVersion to the hub version (v1alpha2.KeptnAppVersion)
+func (src *KeptnAppVersion) ConvertTo(dstRaw conversion.Hub) error {
+	dst, ok := dstRaw.(*v1alpha2.KeptnAppVersion)
+
+	if !ok {
+		return fmt.Errorf("cannot cast KeptnAppVersion to v1alpha2. Got type %T", dstRaw)
+	}
+
+	// Copy equal stuff to new object
+	// DO NOT COPY TypeMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec.Version = src.Spec.Version
+	for _, srcWl := range src.Spec.Workloads {
+		dst.Spec.Workloads = append(dst.Spec.Workloads, v1alpha2.KeptnWorkloadRef{
+			Name:    srcWl.Name,
+			Version: srcWl.Version,
+		})
+	}
+	dst.Spec.PreDeploymentTasks = src.Spec.PreDeploymentTasks
+	dst.Spec.PostDeploymentTasks = src.Spec.PostDeploymentTasks
+	dst.Spec.PreDeploymentEvaluations = src.Spec.PreDeploymentEvaluations
+	dst.Spec.PostDeploymentEvaluations = src.Spec.PostDeploymentEvaluations
+	// Set sensible defaults for new fields
+	dst.Spec.Revision = 1
```

Comment:
I would move this after the copy the old data into the new object, i.e., at ~line 65